### ISSUE TITLE
CE2: Code Minimap has unexpected scrolling behavior

### ISSUE THREAD

**Issue by TheGitMagician** [YoYoGames/GameMaker-Bugs#5294]
### Description

The code minimap is one of the great new additions to CE2. However, right now it doesn't really do its job.

One of the main ideas of a code minimap is that **it shouldn't scroll as long as it can display all of the code**. That way you can get a static overview where in a longer piece of code you are located and can jump around in it.

Currently, the minimap in CE2 starts scrolling even if it has lots of space left to display the whole script. Actually the minimap doesn't scroll independently from the main scrollbar - but acts exactly like the scrollbar. **When you click on the minimap somewhere you don't actually jump to that piece of code but you jump to that y-position of the main scrollbar**.

For comparison I have recorded two GIF videos. The first shows Visual Studio Code's code minimap. Notice two things:
1. The minimap never scrolls and stays static because it has enough space to display all of the code
2. The minimap scrolls independently from the main scrollbar. It shows the section of code currently highlighted on it, whereas the main scrollbar shows the actual y-position in the document.

![vscode](https://github.com/YoYoGames/GameMaker-Bugs/assets/25714975/e57736f3-8f30-4331-ac93-745933c5021b)

In contrast the second GIF shows Gamemaker's minimap: Notice two things:
1. The minimap scrolls immediately even though there is more than enough space to display all of the code. It moves the code around and you lose the ability to see where exactly in your whole script you are located.
2. The minimap doesn't scroll independently from the main scrollbar - when you click on it and drag up and down it basically acts as an extension of the main scrollbar.

![gms](https://github.com/YoYoGames/GameMaker-Bugs/assets/25714975/ca015b4d-2b75-42a0-8086-9b31e644d36a)

[contactus.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14858761/contactus.zip)

### Expected Change

Basically: Make the Minimap behave like in other editors (e.g. VS Code)

### Steps To Reproduce

See above.

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.400 (Betas)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by caitlinrmcintyre** [comment]
Verified fix in IDE v2024.400.0.541  Runtime v2024.400.0.560, thanks.
